Std/Data/AssocList.lean

```diff
@@ -43,6 +43,18 @@ def isEmpty : AssocList α β → Bool
 @[simp] theorem isEmpty_eq (l : AssocList α β) : isEmpty l = l.toList.isEmpty := by
   cases l <;> simp [*, isEmpty, List.isEmpty]
 
+/-- The number of entries in an `AssocList`. -/
+def length (L : AssocList α β) : Nat :=
+  match L with
+  | .nil => 0
+  | .cons _ _ t => t.length + 1
+
+@[simp] theorem length_nil : length (nil : AssocList α β) = 0 := rfl
+@[simp] theorem length_cons : length (cons a b t) = length t + 1 := rfl
+
+theorem toList_length (l : AssocList α β) : l.toList.length = l.length := by
```

Comment:
```suggestion
theorem length_toList (l : AssocList α β) : l.toList.length = l.length := by
```
ditto for all the other theorems below

Reply:
Are we sure? Should I rename (the already existing) `mapKey_toList` to `toList_mapKey` as well? I really don't like not being able to name things as they appear with dot notation. :-(

Reply:
Well it has lead to inconsistent naming even in this very PR (`length_nil` and `length_cons` differ from the others because you decided not to use dot notation there). I think the current status is that this is a general rule, but if you want to make a specific exception for this function I'm okay with that provided you do it consistently and name the others in postfix too.

Reply:
Interesting.

I would argue that both `length_cons` and `toList_length` are simultaneously desirable: the thing the theorem is "about" should come first, and theorems should mostly be about functions rather than terms.

Hence `length_cons` is preferable to `cons_length`, because this is telling us about the behaviour of the function `length` on the constructors.

Similarly `toList_length` is preferable to `length_toList`, because it is describing a property of the function `toList`.

Reply:
I prefer `length_toList` as well because of the naming convention [mentioned here](https://github.com/leanprover/std4/pull/329#issuecomment-1786193983).

Reply:
Okay. Under mild protest, I have changed all the names in this file.